data/scenarios/Challenges/Ranching/gated-paddock.yaml

```diff
@@ -0,0 +1,165 @@
+version: 1
+name: Pastoral Island
+description: |
+  Start a wool industry from the local fauna.
+creative: false
+objectives:
+  - goal:
+      - |
+        You've homesteaded on a small Pacific island.
+        It's time to gather resources to trade.
+      - |
+        You encounter some feral sheep. They look like a promising
+        source of wool.
+      - |
+        First, paddock the sheep so they don't drown.
+        Construct a fence using the nearby gate as a starting point.
+        Make sure there are no gaps!
+      - |
+        Note that you can use the "drill" command (by way of the "post puller" tool)
+        to demolish a fence that has been "placed".
```

Comment:
One cannot walk on top of the fence, and `grab` requires the robot to be on the same tile as the thing being grabbed.

Reply:
@kostmo that is not true, I remember I specifically allowed system robots to do that:
https://github.com/swarm-game/swarm/blob/29594abfa23e0ad10e36267a4192443f1a0aea0c/src/Swarm/Game/Step.hs#L1676-L1677

Reply:
But it is the `base` robot that may need to dismantle the fence, not a system robot.

Reply:
Oh, sorry I agree that drill is necessary (now I see why). I just wanted to object to the statement because I thought you were responding to my other comments, where walking on the fence is relevant. 😅

Reply:
Perhaps you could add a program to the base robot which just `def`ines a new command like `dismantle` which is just a synonym for drill?

Reply:
> Perhaps you could add a program to the base robot which just `def`ines a new command like `dismantle` which is just a synonym for drill?

That could be nice.  Filed #838; currently if add the following to `base` in the scenario file:
```
    program: |
      def dismantle = drill; end;
```
When I try to `dismantle` in the REPL, I get this error:
```
Fatal error: Reference to unknown memory cell 0
```

Reply:
oh no :scream: